tests/unit/customizations/history/test_db.py

```diff
@@ -0,0 +1,711 @@
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'CLI_ARGUMENTS', 1234, '["s3", "ls"]')
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+        self.assertEqual(request_id, '')
+
+    def test_can_write_api_call_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'API_CALL',
+            'payload': {
+                'service': 's3',
+                'operation': 'ListBuckets',
+                'params': {},
+            },
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'API_CALL', 1234, json.dumps({
+                'service': 's3',
+                'operation': 'ListBuckets',
+                'params': {},
+            }))
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+        self.assertTrue(self.UUID_PATTERN.match(request_id))
+
+    def test_can_write_http_request_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'HTTP_REQUEST',
+            'payload': {
+                'method': 'GET',
+                'headers': {},
+                'body': '...',
+            },
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'HTTP_REQUEST', 1234, json.dumps({
+                'method': 'GET',
+                'headers': {},
+                'body': '...',
+            }))
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+
+    def test_can_write_http_response_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'HTTP_RESPONSE',
+            'payload': {
+                'streaming': False,
+                'headers': {},
+                'body': '...',
+                'status_code': 200,
+                'request_id': '1234abcd'
+            },
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'HTTP_RESPONSE', 1234, json.dumps({
+                'streaming': False,
+                'headers': {},
+                'body': '...',
+                'status_code': 200,
+                'request_id': '1234abcd'
+            }))
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+
```

Comment:
Do we want to assert for uid patttern on request id's?

Reply:
I can't unless there was a prior API_CALL made. I can either change that logic or have an API_CALL be made first for all these tests.

Reply:
Ah I think we should just then at least for one of the `API_CALL` tests there is a pattern matching assertion for request_id if it is not already there.